src/stac_utils/google.py

```diff
@@ -186,6 +190,45 @@ def auth_sheets(
     )
 
 
+def auth_drive(
+    scopes: list[str] = None,
+    cache_discovery: bool = False,
+    service_account_blob: dict = None,
+    service_account_env_name: str = "SERVICE_ACCOUNT",
+    subject: str = None,
+    **kwargs,
+) -> Resource:
+    """
+    Returns an initialized Drive client object
+
+    :param scopes: Desired scopes, defaults to `[drive]`
+    :param cache_discovery: `False` unless specified. If cache discovery is desired, set to `True`.
+    :param service_account_blob: Service account blob
+    :param service_account_env_name: Environmental variable name for service account
+    :param subject: Service account subject
+    :return: Drive client object
+    """
+
+    if "is_auto_credential" in kwargs:
+        kwargs.pop("is_auto_credential")
+
+    scopes = scopes or ["drive"]
+    credentials = get_credentials(
+        scopes=scopes,
+        service_account_blob=service_account_blob,
+        service_account_env_name=service_account_env_name,
+        subject=subject,
+    )
+
+    return build(
+        "drive",
+        "v3",
+        credentials=credentials,
+        cache_discovery=cache_discovery,
+        **kwargs,
+    )
```

Comment:
optional (might want to consult with micheál to get his opinion on whether this should happen) but this feels pretty repeated from auth_sheets and auth_gmail - i wonder if we should pull out the repeated code into something and then just set scopes/serviceName/version